### PR TITLE
[Android text input] fix android autofill on focused text field

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
Fixes https://github.com/flutter/flutter/issues/75833

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [ ] All existing and new tests are passing.
- [ ] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

File: shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java [hunk 1]
```diff
@@ -418,6 +418,8 @@ private void setPlatformViewTextInputClient(int platformViewId) {
     mRestartInputPending = false;
   }
 
+  // Called by the text input channel to update the text input plugin with the
+  // latest TextEditState from the framework.
```
Comment (justinmc) [inline shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java:422]: Nit: TextEditState => TextEditingState

File: shell/platform/android/test/io/flutter/plugin/editing/TextInputPluginTest.java [hunk 2]
```diff
@@ -927,6 +929,92 @@ public void autofill_testLifeCycle() {
...
+
+    // Autofill both fields.
+    textInputPlugin.autofill(autofillValues);
+
+    // Verify the Editable has been updated.
+    assertTrue(textInputPlugin.getEditable().toString().equals("focused field"));
+
+    // The autofill value of the focused field is sent via updateEditingState.
+    verify(textInputChannel, times(1))
+        .updateEditingState(anyInt(), eq("focused field"), eq(13), eq(13), eq(-1), eq(-1));
```
Comment (justinmc) [inline shell/platform/android/test/io/flutter/plugin/editing/TextInputPluginTest.java:1008]: Where do the 13s come from?

**Review by justinmc — APPROVED** [review]
LGTM 👍

File: shell/platform/android/test/io/flutter/plugin/editing/TextInputPluginTest.java (hunk 2 above)
Comment (LongCatIsLooong) [reply shell/platform/android/test/io/flutter/plugin/editing/TextInputPluginTest.java:1008]: `"focused field".length == 13`, the autofill implementation puts the caret at the end.

File: shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java (hunk 1 above)
Comment (LongCatIsLooong) [reply shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java:422]: The class that this method takes as a parameter is called `TextEditState`.

**Pull request merged**